analyses/org.eclipse.tracecompass.incubator.lttng2.ust.extras.core/src/org/eclipse/tracecompass/incubator/internal/lttng2/ust/extras/core/pthread/PThreadLockGraphHandler.java

```diff
@@ -131,23 +132,26 @@ private void handleAcquireLockEvent(ITmfEvent event) {
         OsWorker worker = getOrCreateKernelWorker(event, tid);
 
         // Get the vertex for the last request
-        TmfVertex lastReqVertex = fLastRequest.get(tid, fieldValue);
+        ITmfVertex lastReqVertex = fLastRequest.get(tid, fieldValue);
         if (lastReqVertex == null) {
             return;
         }
 
         // Get the last lock owner
         LastLockOwner lastOwner = fLastLockOwner.get(fieldValue);
-        if (lastOwner != null && lastOwner.fTid != tid && lastOwner.fVertex.getTs() > lastReqVertex.getTs()) {
+        if (lastOwner != null && lastOwner.fTid != tid && lastOwner.fVertex.getTimestamp() > lastReqVertex.getTimestamp()) {
             // This thread has been blocked, add the proper vertices and links
-            TmfGraph graph = Objects.requireNonNull(fProvider.getAssignedGraph());
+            ITmfGraph graph = Objects.requireNonNull(fProvider.getGraph());
+
             // First add a vertex at the time of lock request
-            graph.append(worker, lastReqVertex, EdgeType.RUNNING);
+            graph.append(lastReqVertex, new OSEdgeContextState(OSEdgeContextEnum.RUNNING));
             // Then add the blocked transition for the current worker
-            TmfVertex unblockVertex = new TmfVertex(event.getTimestamp().toNanos());
-            graph.append(worker, unblockVertex, EdgeType.BLOCKED);
+            ITmfVertex unblockVertex = graph.createVertex(worker, event.getTimestamp().toNanos());
+            graph.append(unblockVertex, new OSEdgeContextState(OSEdgeContextEnum.BLOCKED));
             // And add the vertical link between the unlock and the acquisition
-            lastOwner.fVertex.linkVertical(unblockVertex);
+            // TODO: check if it's the correct replacement
+//            lastOwner.fVertex.linkVertical(unblockVertex);
+            graph.edgeVertical(lastOwner.fVertex, unblockVertex, new OSEdgeContextState(OSEdgeContextEnum.DEFAULT), MUTEX_FIELD);
```

Comment:
I'm not sure if this is a correct replacement.

Reply:
Looks about right to me. @arfio  please confirm!

Reply:
Seems good to me also.